### PR TITLE
Test transcript segment validation and remove validate_stm

### LINKED ISSUE #105: validate_stm should be exposed at the command line
**Describe the solution you'd like**
validate_stm should be a command line tool

**Describe alternatives you've considered**
Users can invoke this by writing python code

**Additional context**
Requested by users of asrtoolkit

### PR DESCRIPTION
**Pull request by cbrochtrup**
Test our transcript validation and `convert_transcript` is the same function and it's confusing for users.

Other fixes along the way:
- Make segments with start > stop invalid
- Add test converting transcripts with invalid lines.
- Remove validate_stm since it's a duplicate of convert

Fixes #105

### CODE REVIEW

File: samples/invalid.stm [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+unknown 2 UnknownSpeaker 0.00 3.20 <o,f0,male> testing testing, one two three!
+
+Invalid
+wpdofin aposifn209r8n3- 938r1n-30912n- -asdf- -13r12412mmv.
+unknown 1 UnknownSpeaker 0.00 3.20 31 <o,f0,male> testing testing one two three
+unknown 1 UnknownSpeaker 0.00 3.20 <o,f0,male,garbage> testing testing one two three
+unknown 1 typo UnknownSpeaker 0.00 3.20 <o,f0,male,garbage> testing testing one two three
+unknown 2 UnknownSpeaker 0.00 3.20 <o,f0,alien> testing testing one two three
+unknown 2 UnknownSpeaker 0.00 3.20 <new_letter,f0,alien> testing testing one two three
+unknown 2 UnknownSpeaker 9.00 3.20 <o,f0,female> testing testing one two three
```
Comment (sv-github) [inline samples/invalid.stm:13]: :+1: for tests

`typo` :smile:

File: tests/test_remove_invalid_lines.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    validate_sample('.stm', EXPECTED_UNFORMATTED_TRANSCRIPTS, 3)
+
+
+def test_stm_to_json():
+    " Test stm to gk json validation "
+    validate_sample('.json', EXPECTED_FORMATTED_TRANSCRIPTS, 3)
+
+
+def test_stm_to_vtt():
+    " Test stm to gk json validation "
```
Comment (sv-github) [inline tests/test_remove_invalid_lines.py:42]:
```suggestion
    " Test stm to gk vtt validation "
```

File: tests/test_remove_invalid_lines.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    validate_sample('.vtt', EXPECTED_FORMATTED_TRANSCRIPTS, 3)
+
+
+def test_stm_to_html():
+    " Test stm to gk json validation "
```
Comment (sv-github) [inline tests/test_remove_invalid_lines.py:47]:
```suggestion
    " Test stm to html validation "
```

File: tests/test_remove_invalid_lines.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+"""
+Test wer calculation
```
Comment (sv-github) [inline tests/test_remove_invalid_lines.py:3]:
```suggestion
Test invalid line removal
```

File: samples/invalid.stm [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Valid
+unknown 1 UnknownSpeaker 0.00 3.20 <o,f0,male> testing testing one two three or maybe 10 4
+unknown chan UnknownSpeaker 0.00 3.20 <o,f0,male> testing testing one two three
+unknown 2 UnknownSpeaker 0.00 3.20 <o,f0,male> testing testing, one two three!
```
Comment (sv-github) [inline samples/invalid.stm:4]: if you had an invalid line up here (on line2, line3 or line4), would the `convert()` def stop at the first invalid line?

should we check for valid lines in the entire file or do we just stop converting at the first invalid occurrence and then come back to the file?

Comment (cbrochtrup) [reply samples/invalid.stm:4]: Yup, the line order doesn't matter so I separated them for ease of understanding.

**Review by sv-github — APPROVED** [review]

**Pull request merged**